File: android/app/src/main/java/com/lightningkite/kwifttemplate/shared/views/ViewPagerDemoVG.kt
package com.lightningkite.kwifttemplate.shared.views

import android.view.View
import com.lightningkite.kwift.observables.actual.bind
import com.lightningkite.kwift.observables.shared.ObservableStack
import com.lightningkite.kwift.observables.shared.StandardObservableProperty
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwift.views.shared.ViewGenerator
import com.lightningkite.kwifttemplate.xml.RowTestXml
import com.lightningkite.kwifttemplate.xml.ViewPagerDemoXml

class ViewPagerDemoVG(val stack: ObservableStack<ViewGenerator>) : ViewGenerator() {
    override val title: String get() = "View Pager Demo"

    val items: List<String> = listOf(
        "First",
        "Second",
        "Third"
    )
    val selectedIndex: StandardObservableProperty<Int> = StandardObservableProperty(0)

    override fun generate(dependency: ViewDependency): View {
        val xml = ViewPagerDemoXml()
        val view = xml.setup(dependency)

        xml.viewPager.bind(items, selectedIndex) { it ->
            val xml = RowTestXml()
            val view = xml.setup(dependency)
            xml.label.text = it
            return@bind view
        }
        xml.viewPagerIndicator.bind(items.size, selectedIndex)

        return view
    }
}

File: ios/Kwift Template/com/lightningkite/kwifttemplate/shared/views/ViewPagerDemoVG.swift
//Package: com.lightningkite.kwifttemplate.shared.views
//Converted using Kwift2

import Foundation
import Kwift
import KwiftMaps



public class ViewPagerDemoVG: ViewGenerator {
    
    public var stack: ObservableStack<ViewGenerator>
    
    override public var title: String {
        get {
            return "View Pager Demo"
        }
    }
    public var items: Array<String>
    public var selectedIndex: StandardObservableProperty<Int32>
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = ViewPagerDemoXml()
        var view = xml.setup(dependency)
        xml.viewPager.bind(items, selectedIndex) { (it) in 
            var xml = RowTestXml()
            var view = xml.setup(dependency)
            xml.label.text = it
            return view
        }
        xml.viewPagerIndicator.bind(items.size, selectedIndex)
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    public init(stack: ObservableStack<ViewGenerator>) {
        self.stack = stack
        let items: Array<String> = ["First", "Second", "Third"]
        self.items = items
        let selectedIndex: StandardObservableProperty<Int32> = StandardObservableProperty(0)
        self.selectedIndex = selectedIndex
        super.init()
    }
    convenience public init(_ stack: ObservableStack<ViewGenerator>) {
        self.init(stack: stack)
    }
}
 
